ektelo/client/inference.py

```diff
@@ -116,53 +70,24 @@ class ScalableInferenceOperator(InferenceOperator):
 
     def _apply_scales(self, Ms, ys, scale_factors):
         if scale_factors is None:
-            M = Ms
-            y = ys
-            noise_scales = [1.0] * len(y)
-        else:
-            assert type(Ms) == list and type(ys) == list
-            assert len(Ms) > 0 and len(Ms) == len(ys) and len(ys) == len(scale_factors)
+            if type(Ms) is list:
+                return matrix.VStack(Ms), np.concatenate(ys)
+            return Ms, ys
+        assert type(Ms) == list and type(ys) == list
+        assert len(Ms) > 0 and len(Ms) == len(ys) and len(ys) == len(scale_factors)
+
+        A = matrix.VStack([M*(1.0/w) for M, w in zip(Ms, scale_factors)])
+        y = np.concatenate([y/w for y, w in zip(ys, scale_factors)])
         
-            M = np.empty((0, np.product(Ms[0].shape[1])), int)
-            y = []
-            noise_scales = []
-
-            for i in range(len(scale_factors)):
-                M = sparse.vstack((M, Ms[i]))
-                y = np.concatenate((y, ys[i]))
-                noise_scales = np.concatenate((noise_scales, [scale_factors[i]] * len(ys[i])))
-
-        return get_A(M, noise_scales), get_y(y, noise_scales).flatten()
-
+        return A, y
 
 class LeastSquares(ScalableInferenceOperator):
 
-    def __init__(self, method='lsmr', l2_reg=0.0, known_total=None, stein=False):
```

Comment:
Here and elsewhere, are we pretty sure nothing will want to set these params downstream?

Reply:
I think so, stein was an experimental thing that didn't really make a difference.  known_total is a useful feature, which we should consider bringing back but I think the implementation there is also experimental and may not be as robust as lsmr.